tests/Microsoft.Identity.Test.UIAutomation/XamarinDroidMSALTests.cs

```diff
@@ -64,9 +68,66 @@ public void InitializeBeforeTest()
         }
 
         /// <summary>
-        /// Runs through the standard acquire token flow, using the default app configured UiBehavior = Login
+        /// Test runner to run all tests, as test initialization is expensive.
         /// </summary>
         [Test]
+        public void RunAllTests()
+        {
+            var tests = new List<Action>()
+            {
+                AcquireTokenTest,
+                AcquireTokenSilentTest,
+
+                PromptBehavior_Consent_SelectAccount,
+
+                AcquireTokenADFSV3InteractiveFederatedTest,
+                AcquireTokenADFSV3InteractiveNonFederatedTest,
+                AcquireTokenADFSV4InteractiveFederatedTest,
+                AcquireTokenADFSV4InteractiveNonFederatedTest,
+
+                B2CFacebookProviderEditPolicyAcquireTokenTest,
+                B2CFacebookProviderWithB2CLoginAuthorityAcquireTokenTest,
+                B2CFacebookProviderWithMicrosoftAuthorityAcquireTokenTest,
+                //B2CGoogleProviderWithB2CLoginAuthorityAcquireTokenTest,
+                //B2CGoogleProviderWithMicrosoftAuthorityAcquireTokenTest,
+                B2CLocalAccountAcquireTokenTest,
+                B2CLocalAccountAcquireTokenWithMicrosoftAuthorityTest,
+            };
+
+            var hasFailed = false;
+            var stringBuilderMessage = new StringBuilder();
+
+            foreach (Action test in tests)
+            {
+                try
+                {
+                    LogMessage($"Running test: {test.Method.Name}", stringBuilderMessage);
+                    test();
+                }
+                catch(Exception ex)
```

Comment:
How will this behave with the Assert.Fail() in the test code? will they be caught here?